### ISSUE TITLE
Add configuration support for disabling transactions with JDBC session store

### ISSUE THREAD

**Issue by vpavic** [spring-projects/spring-session#1469]
We added support for transaction-less mode for `JdbcOperationsSessionRepository` in #1046 however to use this feature users must register `JdbcOperationsSessionRepository` manually using an appropriate constructor.

We should provide first-class configuration support for this, especially since with Spring Boot registering `SessionRepository` bean manually disables auto-configuration.

**Comment by vpavic** [comment]
This is now resolved by adding configuration support for picking up `TransactionOperations` bean named `springSessionTransactionOperations`. Coupled with the new transactionless `TransactionOperations` factory that will be available in Spring 5.2, disabling transactions for `JdbcOperationsSessionRepository` is as simple as:

```java
@Bean
public TransactionOperations springSessionTransactionOperations() {
	return TransactionOperations.withoutTransaction();
}
```
```

**Comment by gotson** [comment]
> This is now resolved by adding configuration support for picking up `TransactionOperations` bean named `springSessionTransactionOperations`. Coupled with the new transaction-less `TransactionOperations` factory that will be available in Spring Framework 5.2, disabling transactions for `JdbcOperationsSessionRepository` is as simple as:
> 
> ```java
> @Bean
> public TransactionOperations springSessionTransactionOperations() {
>     return TransactionOperations.withoutTransaction();
> }
> ```
> 
> This approach also allows to customize transactional behavior in any other desired way.

Sorry for commenting on a closed issue, but I had to scour the code, issues, and project reference page multiple times until I could find this. SQLite performance using transactions is very bad, because it only supports `SERIALIZABLE` level of transactions.

Would it be possible to add this in the reference documentation? That would most likely help other people i'm sure.

**Comment by vpavic** [comment]
@gotson #1011 is probably the issue worth tracking. You could also add a comment in there as well, mentioning the things you find to be missing.
